server/clientRoutes.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const assert = require('bsert');
+const Config = require('bcfg');
+const logger = require('./logger');
+
+function clientsRouter(clients) {
+  const router = express.Router({ mergeParams: true });
+  let token, id, config, reqClients;
+
+  // return object with
+  // relevant info for each client
+  // keyed to the client id
+  const clientInfo = {};
+  clients.forEach(
+    client =>
+      (clientInfo[client.config.str('id')] = {
+        chain: client.config.str('chain', 'bitcoin')
+      })
+  );
+  router.get('/', (req, res) => res.status(200).json(clientInfo));
+
+  // middleware for setting constants based on
+  // the route being used
+  router.use('/:id', (req, res, next) => {
+    id = req.params.id;
+    if (!clients.has(id))
+      return res
+        .status(404)
+        .send(`Sorry, there was no client with the id ${id}`);
```

Comment:
`res.json({ message: '...' })`

Reply:
Also, should these routes be protected by auth?

Reply:
`.send({ error: { message: e.message, code: e.code, type: e.type } });`

This is the bcoin error format